### ISSUE TITLE
LN2_DEVEIN required parameters appear incorrect

### ISSUE THREAD

**Issue by dowdlelt** [layerfMRI/LAYNII#28]
Per the devein blogpost, and the help text of `LN2_DEVEIN` the linear method should only require the BOLD activity input and the layer file. However, if you attempt to run `LN2_DEVEIN` with just `-linear -input BOLD_XX.nii -layer_file layer_XX.nii` you will get an error:
```
LN2_DEVEIN -linear -input lo_BOLD_act.nii -layer_file lo_layers.nii 
 there will be no deconvolution, just layer-dependent depth scaling
** missing option '-column_file'
```
Even if columns, are included (perhaps they are actually useful? not clear), you still see the following:
```
LN2_DEVEIN -linear -input lo_BOLD_act.nii -layer_file lo_layers.nii -column_file lo_columns.nii 
 there will be no deconvolution, just layer-dependent depth scaling
** missing option '-ALF'
```

**Comment by ofgulban** [comment]
Hi @dowdlelt  , thanks a lot for posting this issue. Upon your report:
1. I have reviewed the code and restructured the flow to follow the claim in the blog post: "Linear deveining only requires and input and layer file". Thanks again for recognizing this inconsistency. The current usage for linear scaling is : `LN2_DEVEIN -input BOLD_XX.nii -layer_file layer_XX.nii` as you tried. I have removed the `-linear` flag as the program by default tries linear scaling if `-ALF`, `-CBV` or `-column_file` are not given.
2. I have improved the output naming. Now the program assigns a unique tag based on the selected option (e.g. `devein`, `deveinLinear` or `deveinCBV`) 
3. I have detected a small scaling miscalculation (+1 to layer id's in division part) in linear mode and fixed it.

As a sidenote for @layerfMRI , I think we might also use `metric` files from `LN2_LAYERS` to avoid linear scaling issues in over-or under- layered volumes.

**Comment by layerfMRI** [comment]
Thanks Faruk, 
This is appreciated. 

I have a few questions, though. 

1.) So you say that the program determines, which model to use by the kind of input that is given? This would mean that the description in the Bolg is no longer correct, no? If you don't mind, i would include the (now optional) "-linear" flag again. This way people would still be able to go through the examples as described in the blog. Also, if the "-linear" flag doesn't have any functionality, just so the program can be executed the old way without an error. 

This new way of doing the linear detrending also means that one cannot run the linear mode with the columns file. I am afraid this is not really desirable. Because in this way it is not possible anymore to pool voxels within columns? Without this averaging, the noise characteristics are no longer comparable with the other ways of deveining. Those methods still use the columns file, no? 
People, who are currently using this method will complain that with the new version, everything is noisier? 

2.) I like the new naming convention. for deveinCBV and deveinLinear. I think we should call the third option "deveinDeconv", would you agree? 

3.) I am not sure, if I get what you did here? Since the for loop and the index were shifted by the same +1, it should be the same result. E.g. when I use the lo_layers.nii as input, I get the same result for both versions. In any case, I fully agree with your new way of implementation.

**Comment by ofgulban** [comment]
Thanks @layerfMRI . I would like to point some inconsistencies in the blog post to resolve the remaining issues:

1. See the following caption from Option 1 figure:
> Input and output of LN2_DEVEIN for depth-dependent offset correction. Since the offset correction is not based on column-specific physiological assumptions, **it does not require the column file nor the ALF file to work**. In this form of de-veining, LN2_DEVEIN simply removes the linear trend, while keeping its mean. 

As you can see the description and the code was inconsistent already as the previous implementation did require `column file` and `ALF` file. To address resolve this, I can reintroduce the `--linear` flag with `columns file` so that the blog example command works as before while the program also works as @dowdlelt correctly tried running it without giving the `columns file` and `ALF` file following the description.

2. I agree.
3. I will double check.

**Comment by layerfMRI** [comment]
1. Yes, you are right. My bad. 
I am inconsistent with the blog. 

I like you second solution. I updated the command in the blog few moments ago.

The line now reads, `LN2_DEVEIN -linear -layer_file lo_layers.nii -input lo_BOLD_act.nii`

(Note; I also corrected an inconsistent usage of lo_BOLD_**stat**.nii and changed them all to lo_BOLD_**act**.nii).

I guess this solution will still mean that people will get different results for the -linear option compared to the LAYNII version few weeks ago. Because it is not averages within columns. I am fine with it. 
Maybe we need to mention this in the release notes, though. 
As far as I am concerned, I think @dowdlelt's comment should be fixed now. And this issue can be closed.
